src/switchboard.erl

```diff
@@ -198,19 +198,16 @@ await(Account, Type, Timeout) ->
     Pid.
 
 
-%% @todo implement a pool around active connections, checkout the active process
-%% from this.
 checkout(Account) ->
-    {Active, _} = gproc:await(key_for(Account, active), 5000),
+    Active = poolboy:checkout({via, gproc, key_for(Account, pool)}),
     {ok, Active}.
 
 
-%% @todo return a checked out IMAP connection.
-return(_Account, _IMAP) ->
-    ok.
+return(Account, IMAP) ->
+    ok = poolboy:checkin({via, gproc, key_for(Account, pool)}, IMAP).
```

Comment:
nitpick: could we have this function still return `ok`? I don't think anything depends on it, but it does change the behaviour.